### PR TITLE
take gardient accumulation value into account when calculating total actual training steps used for determining when training is complete

### PR DESCRIPTION
**Pull request by breakcore2**
training_complete is comparing global_step against actual_train_steps to determine if the training is done but actual_train_steps does not represent (number of training images) * (desired epoch) when gradient accumulation != 1 so training is completed earlier than desire based on user input value of epoch size and dataset size.
We multiply actual_train_steps by gradient accumulation amount to arrive back at the correct value.

### CODE REVIEW

**Comment by breakcore2** [comment]
Issue: #627

**Comment by d8ahazard** [comment]
So, I spent a lot of time mucking around with trying to figure out how to come up with a logical step count that took into consideration grad_acc steps, but there just isn't one. The problem is that grad steps are counted when accelerator.sync_gradients is called, and this doesn't necessarily happen as frequently as we set it to. If grad * batch does not come out to the number of instance images we have, then the grad steps will be a different value than specified.

I've implemented "logical" step counting in the latest version so that everything at least aligns with the number of trained epochs. Maybe it's not technically perfect, but it's better than the way it was being counted.
